packages/url-state-provider/src/v2/encode.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import queryString from "query-string"
+import { ObjectToEncode, Primitive } from "./types"
+
+const isPrimitive = (value: Primitive | Primitive[]) => {
+  return (
+    typeof value === "string" ||
+    typeof value === "number" ||
+    typeof value === "boolean" ||
+    value === null ||
+    value === undefined ||
+    value instanceof RegExp
+  )
+}
+
+const validateObjectToEncode = (object: ObjectToEncode) => {
+  if (object === null || typeof object !== "object" || object instanceof RegExp || Array.isArray(object)) {
```

Comment:
what about Your `empty objects` your validation currently allows empty objects (e.g., {})?

Reply:
Good catch, I've fixed it and added a test case for that as well.